controllers/tasks.js

```diff
@@ -328,41 +328,32 @@ const updateTaskStatus = async (req, res, next) => {
 
     if (task.taskNotFound) return res.boom.notFound("Task doesn't exist");
     if (task.notAssignedToYou) return res.boom.forbidden("This task is not assigned to you");
-    if (
-      task.taskData.status === TASK_STATUS.VERIFIED ||
-      status === TASK_STATUS.MERGED ||
-      status === TASK_STATUS.BACKLOG
-    )
+    if (TASK_STATUS.BACKLOG === status) {
       return res.boom.forbidden("Status cannot be updated. Please contact admin.");
-
+    }
     if (userStatusFlag) {
-      if (task.taskData.status === TASK_STATUS.DONE && req.body.percentCompleted < 100) {
-        if (status === TASK_STATUS.DONE || !status) {
-          return res.boom.badRequest("Task percentCompleted can't updated as status is DONE");
+      if (task.taskData.status === TASK_STATUS.DONE) {
+        return res.boom.forbidden("Status cannot be updated. Please contact admin.");
+      }
+    } else {
+      if (task.taskData.status === TASK_STATUS.VERIFIED || TASK_STATUS.MERGED === status) {
+        return res.boom.forbidden("Status cannot be updated. Please contact admin.");
+      }
+      if (task.taskData.status === TASK_STATUS.COMPLETED && req.body.percentCompleted < 100) {
+        if (status === TASK_STATUS.COMPLETED || !status) {
+          return res.boom.badRequest("Task percentCompleted can't updated as status is COMPLETED");
```

Comment:
here you used multiple conditional checks that can be consolidated into reusable functions. eg., you can create separate functions to handle status checks and progress checks.